js/server/process-vs.js

```diff
@@ -14,103 +22,113 @@ function processVSGlobalState (lastGlobalState, timestamp, eventsByUser) {
       })
     )
   );
-  const updatedUsers = _.mapValues(users, user => {
+  const updatedUsers = _.mapValues(users, (user) => {
     return {
       ...user,
-      tickets: user.tickets.map(ticket => {
+      tickets: user.tickets.map((ticket) => {
         const additionalAmount =
           (ticket.amount / (totalShares || 1)) * globalRewardAccrued;
         return {
           ...ticket,
           mul: Math.min(ticket.mul + 0.75 / MULTIPLIER_MATURITY, 1),
-          reward: ticket.reward + additionalAmount
+          reward: ticket.reward + additionalAmount,
         };
-      })
+      }),
     };
   });
   return updatedUsers;
 }
 
-function processUserEvents (users, eventsByUser) {
-  _.forEach(eventsByUser, userEvents => {
+function processUserEvents(users, eventsByUser) {
+  _.forEach(eventsByUser, (userEvents) => {
     const previousUser =
       userEvents.length > 0 && users[userEvents[0].delegateAddress];
     const previousTickets = previousUser ? previousUser.tickets : [];
 
-    userEvents.forEach(event => {
+    userEvents.forEach((event) => {
       const user = users[event.delegateAddress] || {
         tickets: [],
         claimed: 0,
         dispensed: 0,
-        forfeited: 0
+        forfeited: 0,
       };
       users[event.delegateAddress] = user;
-      const validator = users[event.valSifAddress] || {
+      const validator = users[event.validatorSifAddress] || {
         tickets: [],
         claimed: 0,
         dispensed: 0,
-        forfeited: 0
+        forfeited: 0,
       };
-      users[event.valSifAddress] = validator;
-      validator.commisionClaimed = validator.commisionClaimed || 0;
+      users[event.validatorSifAddress] = validator;
+      validator.commissionClaimed = validator.commissionClaimed || 0;
+
+      // Is deposit (adding funds)
       if (event.amount > 0) {
         const isRedelegation =
           userEvents.length > 1 &&
-          userEvents.some(other => {
+          userEvents.some((other) => {
             return other.amount === -event.amount;
           });
-        const redelegatedTicket =
-          isRedelegation &&
-          previousTickets.find(t => t.amount === event.amount);
+        if (
+          !isRedelegation &&
+          // is after deposits are allowed
+          event.timestamp > config.DEPOSITS_ALLOWED_DURATION_MS / 1000 / 60
+        ) {
+          // skip
+          return;
+        }
+        const redelegatedTicket = isRedelegation
+          ? previousTickets.find((t) => t.amount === event.amount)
+          : false;
         const newTicket = {
-          commision: event.commision,
-          valSifAddress: event.valSifAddress,
+          commission: event.commission,
+          validatorSifAddress: event.validatorSifAddress,
           amount: event.amount,
           mul: redelegatedTicket ? redelegatedTicket.mul : 0.25,
           reward: 0,
           timestamp: moment
             .utc(START_DATETIME)
```

Comment:
@nikohosn have the snapshots been updated to start at START_DATETIME, or will they include some entries earlier than that?

if so, then we'll have a situation where START_DATETIME > event.timestamp and so the logic here may be off @mccallofthewild

Reply:
@musnit @mccallofthewild the snapshots have been updated to start at START_DATETIME.